### PR TITLE
Lsh c functions

### PR DESCRIPTION
**Pull request by ADKosm**
#2 issue

### CODE REVIEW

File: lib/lsh--0.0.1.sql [hunk 1]
```diff
@@ -1,7 +1,31 @@
 -- complain if script is sourced in psql, rather than via CREATE EXTENSION
-\echo Use "CREATE EXTENSION lsh_index" to load this file. \quit
+\echo Use "CREATE EXTENSION lsh;" to load this file. \quit
 
-CREATE FUNCTION lsh()
-    RETURNS text
-    AS '$libdir/lsh'
-    LANGUAGE C IMMUTABLE STRICT;
\ No newline at end of file
+CREATE OR REPLACE FUNCTION dot(anyarray, anyarray)
```
Comment (Croolis) [inline lib/lsh--0.0.1.sql]: maybe we should use 'lsh' prefix to identify that function is related to this extension?

File: lib/lsh--0.0.1.sql [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+    RETURNS float4[]
+    AS '$libdir/lsh', 'gauss_vector'
+    LANGUAGE C STRICT IMMUTABLE;
+
+CREATE OR REPLACE FUNCTION uniform_number(float4)
+    RETURNS float4
+    AS '$libdir/lsh', 'uniform_number'
+    LANGUAGE C STRICT IMMUTABLE;
+
+CREATE OR REPLACE FUNCTION hash_i(x float4[], w float4[], b float4, r float4)
```
Comment (Croolis) [inline lib/lsh--0.0.1.sql]: why not just 'hash'?

File: lib/lsh.c [hunk 3]
```diff
@@ -1,15 +1,154 @@
...
+    &s->elems, NULL, &s->nelems);
+
+  return s;
+}
+
+// --------------------------------
+
+/*
+Calculate dot product between two vectors
+Needed for combute <x, w>
```
Comment (Croolis) [inline lib/lsh.c:94]: 'Is used to compute'

File: lib/lsh.c [hunk 4]
```diff
@@ -1,15 +1,154 @@
...
+  }
+
+  pfree(a);
+  pfree(b);
+  PG_RETURN_FLOAT4(result);
+}
+
+/*
+Generate random vector with normal distribution N(0, 1)
+Needed for generate parameter `w`
```
Comment (Croolis) [inline lib/lsh.c]: 'Is used to generate number'

**Review by Croolis — APPROVED** [review]

File: lib/lsh--0.0.1.sql (hunk 2 above)
Comment (ADKosm) [reply lib/lsh--0.0.1.sql]: Полный hash - это вектор из посчитанных "малых" хешей.

**Pull request merged**